src/routes/LocationPage/index.js

```diff
@@ -30,20 +31,36 @@ class ContentWrapper extends React.Component {
     return `/${this.props.location}/${this.props.language}`
   }
 
+  componentDidMount () {
+    this.updateAvailableLanguages(this.props.path)
+  }
+
   componentWillUpdate (nextProps) {
-    const hierarchy = new Hierarchy(nextProps.path)
+    if (nextProps.path !== this.props.path) {
+      this.updateAvailableLanguages(nextProps.path)
+    }
+  }
+
+  updateAvailableLanguages (path) {
+    const hierarchy = new Hierarchy(path)
 
     // Pass data to hierarchy
     const error = hierarchy.build(this.props.pages)
     if (error) {
       return
     }
 
-    this.props.dispatch(setCurrentAvailableLanguages(hierarchy.top()))
+    const currentPage = hierarchy.top()
+    const redirect = (id, language) => `/${this.props.location}/${language}/redirect?id=${id}`
+    const currentAvailableLanguages = reduce(currentPage.availableLanguages, (acc, id, language) => {
+      acc[language] = redirect(id, language)
+      return acc
+    }, {})
```

Comment:
How can I rewrite that?
I want to map `{ 'de': 12042, 'en': 2131 }` into `{ 'de': '/redirected/path', 'en': '/redirected/path/2' }`